org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/DefaultAsyncHttpClientManager.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.shared.http.async;
+
+import java.io.IOException;
+
+import org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient;
+import org.apache.hc.client5.http.impl.async.HttpAsyncClientBuilder;
+import org.apache.hc.client5.http.impl.nio.PoolingAsyncClientConnectionManagerBuilder;
+import org.eclipse.scout.rt.platform.ApplicationScoped;
+import org.eclipse.scout.rt.platform.IPlatform.State;
+import org.eclipse.scout.rt.platform.IPlatformListener;
+import org.eclipse.scout.rt.platform.PlatformEvent;
+import org.eclipse.scout.rt.platform.exception.ProcessingException;
+
+/**
+ * <p>
+ * Acts as factory and cache for {@link CloseableHttpAsyncClient} (client is initialized only once).
+ * </p>
+ * <p>
+ * Also acts a {@link IPlatformListener}, listening for platform change events. In case of a
+ * {@link State#PlatformStopping} event, the underlying {@link CloseableHttpAsyncClient} is shut down.
+ * </p>
+ */
+@ApplicationScoped
+public class DefaultAsyncHttpClientManager implements IPlatformListener {
+
+  /**
+   * Is {@link DefaultAsyncHttpClientManager} still active?
+   */
+  private volatile boolean m_active = true;
+
+  /**
+   * Initialized?
+   */
+  private volatile boolean m_initialized = false;
+
+  /**
+   * Cached {@link CloseableHttpAsyncClient}.
+   */
+  private volatile CloseableHttpAsyncClient m_client;
+
+  public CloseableHttpAsyncClient getClient() {
+    init();
+    return m_client;
+  }
+
+  /**
+   * Initialize the manager (if not initialized yet). Method call should be cheap as this method is called plenty of
+   * times.
+   */
+  protected void init() {
+    if (!m_initialized) {
+      initSynchronized();
+      m_initialized = true;
+    }
+  }
+
+  /**
+   * Create the {@link CloseableHttpAsyncClient} (using outer factory), fill {@link #m_client} field.
+   */
+  protected synchronized void initSynchronized() {
+    if (m_initialized || !m_active) {
+      return;
+    }
+
+    HttpAsyncClientBuilder builder = createBuilder();
+    interceptCreateClient(builder);
+
+    builder.setConnectionManager(PoolingAsyncClientConnectionManagerBuilder
+        .create()
+        .useSystemProperties()
```

Comment:
Yes, but w/o `useSystemProperties` the `javax.net.ssl.SSLContext.getDefault()` holding our trust manager is not used; in this case it would be necessary to specify trust manager explicitly here.

Reply:
Comment this method is used to use system trust manager.

Reply:
Added comment and removed one call in 912450f97f0febfebac81326d3b7f89eb4bdca12